go/cmd/dolt/commands/push.go

```diff
@@ -228,7 +228,7 @@ func pullerProgFunc(pullerEventCh chan datas.PullerEvent) {
 	}
 }
 
-func progFunc(progChan chan datas.PullProgress) {
+func progFunc(ctx context.Context, progChan chan datas.PullProgress) {
```

Comment:
You changed this interface, but didn't change the behavior here (still relies on timeouts or channel close to exit the loop)

Add some TODOs?

Reply:
i don't understand how to properly drain channels if we short circuit. maybe doesn't matter in the cli because the process won't be long-running, i'll leave a commented block with the `<-ctx.Done(): return`